src/PrivateDns/PrivateDns/Records/GetAzurePrivateDnsRecordSet.cs

```diff
@@ -0,0 +1,110 @@
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The zone in which to create the record set (without a terminating dot).", ParameterSetName = FieldsParameterSetName)]
+        [ResourceNameCompleter("Microsoft.Network/privateDnsZones", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string ZoneName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The DnsZone object representing the zone in which to create the record set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PSPrivateDnsZone Zone { get; set; }
+
+        [Parameter(ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ResourceIdCompleter("Microsoft.Network/privateDnsZones")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The name of the records in this record set (relative to the name of the zone and without a terminating dot).")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The type of DNS records in this record set.")]
+        [ValidateNotNullOrEmpty]
+        public RecordType? RecordType { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string zoneName = null;
+            string resourceGroupName = null;
+
+            switch (this.ParameterSetName)
+            {
+                case FieldsParameterSetName:
+                    zoneName = this.ZoneName;
+                    resourceGroupName = this.ResourceGroupName;
+                    break;
+                case ResourceParameterSetName:
+                    PrivateDnsUtils.GetResourceGroupNameAndZoneNameFromResourceId(this.ParentResourceId, out resourceGroupName, out zoneName);
+                    break;
+                default:
+                    zoneName = this.Zone.Name;
+                    resourceGroupName = this.Zone.ResourceGroupName;
+                    break;
+            }
+
+            if (zoneName != null && zoneName.EndsWith("."))
+            {
+                zoneName = zoneName.TrimEnd('.');
+                this.WriteWarning($"Modifying zone name to remove terminating '.'.  Zone name used is \"{zoneName}\".");
+            }
+
+            if (this.Name != null)
+            {
+                if (this.RecordType == null)
+                {
+                    throw new PSArgumentException("If you specify the Name parameter you must also specify the RecordType parameter.");
```

Comment:
Please fix this using parameter sets instead:

    1. -ResourceGroupName <String> 
       -ZoneName <String>
       [-RecordType <RecordType>]
       [-DefaultProfile <IAzureContextContainer>]
       [<CommonParameters>]

    2. -ParentObject <PrivateDnsZone> 
        [-RecordType <RecordType>]
        [-DefaultProfile <IAzureContextContainer>]
        [<CommonParameters>]

    3. -ParentResourceId <String> 
        [-RecordType <RecordType>]
        [-DefaultProfile <IAzureContextContainer>]
        [<CommonParameters>]

    4. -ResourceGroupName <String> 
       -ZoneName <String>
       -Name <String>
       -RecordType <RecordType>
       [-DefaultProfile <IAzureContextContainer>]
       [<CommonParameters>]

    5. -ParentObject <PrivateDnsZone> 
        -Name <String>
        -RecordType <RecordType>
        [-DefaultProfile <IAzureContextContainer>]
        [<CommonParameters>]

    6. -ParentResourceId <String> 
        -Name <String>
        -RecordType <RecordType>
        [-DefaultProfile <IAzureContextContainer>]
        [<CommonParameters>]

Reply:
changed the file to match this.